### ISSUE TITLE
Organization AccountStore is Null

### ISSUE THREAD

**Issue by mrioan** [stormpath/stormpath-sdk-java#456]
While iterating over a list of `AccountStores` we will get null if one of them is `Organization`. This code will fail (throw an Exception):
```
for (AccountStoreMapping accountStoreMapping : application.getAccountStoreMappings()) {
            AccountStore accountStore = accountStoreMapping.getAccountStore();
                accountStore.accept(new AccountStoreVisitor() {
                    @Override
                    public void visit(Group group) {
                        System.out.println("This account store is a Group: " + group.getName());
                    }

                    @Override
                    public void visit(Directory directory) {
                        System.out.println("This account store is a Directory: " + directory.getName());
                    }

                    @Override
                    public void visit(Organization organization) {
                        System.out.println("This account store is an Organization: " + organization.getName());
                    }
                });
            }
        }
```

Proposed solution:

`AbstractAccountStoreMapping#getAccountStore()` must contain 
```
...
} else if (href.contains("/organizations/")) {
    accountStore = getDataStore().getResource(href, Organization.class);
}
...
```

We also need to add a test to catch this

**Comment by maranai** [comment]
@mrioan I think this issue duplicates https://github.com/stormpath/stormpath-sdk-java/issues/433, which was fixed with PR #441

**Comment by mrioan** [comment]
Closing as it is a duplicate of https://github.com/stormpath/stormpath-sdk-java/issues/433
